data/src/client.rs

```diff
@@ -845,6 +897,51 @@ impl Client {
                     self.registration_required_channels.push(channel.clone());
                 }
             }
+            Command::Numeric(RPL_ISUPPORT, args) => {
+                let args_len = args.len();
+                args.iter().enumerate().skip(1).for_each(|(index, arg)| {
+                    let isupport_operation = arg.parse::<isupport::Operation>();
+
+                    match isupport_operation {
+                        Ok(isupport_operation) => {
+                            match isupport_operation {
+                                isupport::Operation::Add(isupport_parameter) => {
+                                    if let Some(kind) = isupport_parameter.kind() {
+                                        log::info!(
+                                            "[{}] adding ISUPPORT parameter: {:?}",
+                                            self.server,
+                                            isupport_parameter
+                                        );
+                                        self.isupport_parameters.insert(kind, isupport_parameter);
+                                    }
+                                }
+                                isupport::Operation::Remove(_) => {
+                                    if let Some(kind) = isupport_operation.kind() {
+                                        log::info!(
+                                            "[{}] removing ISUPPORT parameter: {:?}",
+                                            self.server,
+                                            kind
+                                        );
+                                        self.isupport_parameters.remove(&kind);
+                                    }
+                                }
+                            };
+                        }
+                        Err(error) => {
+                            if index != args_len - 1 {
+                                log::debug!(
+                                    "[{}] unable to parse ISUPPORT parameter: {} ({})",
+                                    self.server,
+                                    arg,
+                                    error
```

Comment:
Nit: We know we're dealing with `isupport` here, we can shorten some vars: `isupport_operation -> operation` / `isupport_parameter -> parameter`.

Also, we can just call `self.isupport_parameters -> self.isupport`, I think this will reduce a lot of visual redundancy / noise.

Reply:
Sounds good to me!  I tend toward increased verbosity, so the guidance is greatly appreciated. :)

Reply:
I'm definitely all for fully typed names (parameters > params, etc) but here this is just redundant namespacing we don't need since it's implied in the context we're working in imo. At least that's my reasoning